src/client/src/Client.c

```diff
@@ -171,11 +171,17 @@ STATUS createKinesisVideoClient(PDeviceInfo pDeviceInfo, PClientCallbacks pClien
     PKinesisVideoClient pKinesisVideoClient = NULL;
     PStateMachine pStateMachine = NULL;
     BOOL tearDownOnError = TRUE;
-    UINT32 allocationSize, heapFlags, tagsSize;
+    UINT32 allocationSize, heapFlags, tagsSize, logLevel = DEFAULT_LOGGER_LOG_LEVEL;
 
     // Check the input params
     CHK(pDeviceInfo != NULL && pClientHandle != NULL, STATUS_NULL_ARG);
 
+    // Set the log level immediately so that initialization logs are shown

```

Comment:
/nit formatting

```
logLevel = (pDeviceInfo->version >= 1 && 
            pDeviceInfo->clientInfo.loggerLogLevel > 0 && 
            pDeviceInfo->clientInfo.loggerLogLevel <= LOG_LEVEL_PROFILE) 
            ? pDeviceInfo->clientInfo.loggerLogLevel 
            : logLevel;
```

Reply:
Probably can re-look at the rules later. i would like to use the 3 lines separately for the condition